pkg/segment/writer/metrics/metricssegment.go

```diff
@@ -531,6 +554,11 @@ func EncodeDatapoint(mName []byte, tags *TagsHolder, dp float64, timestamp uint3
 			return err
 		}
 	}
+	err = mSeg.mBlock.appendToWALBuffer(walDataPoint)
```

Comment:
a better way may be to expose a func : `mSeg.mBlock.addDpToWal(timestamp, dp, tsid)`
that way you dont have to allocate this `wal.WalDatapoint` on stack.